core/http/src/main/java/org/trellisldp/http/impl/GetHandler.java

```diff
@@ -355,29 +356,22 @@ private ResponseBuilder getLdpNr(final ResponseBuilder builder) {
 
         // Short circuit HEAD requests
         if (HEAD.equals(getRequest().getMethod())) {
-            return builder;
+            return completedFuture(builder);
         }
 
         // Stream the binary content
-        final StreamingOutput stream = new StreamingOutput() {
-            @Override
-            public void write(final OutputStream out) throws IOException {
-                // TODO -- with JDK 9 use InputStream::transferTo instead of IOUtils::copy
-                try (final InputStream binary = getBinaryStream(dsid, getRequest()).toCompletableFuture().join()) {
-                    IOUtils.copy(binary, out);
-                }
-            }
-        };
-
-        return builder.entity(stream);
+        // TODO -- with JDK 9 use InputStream::transferTo instead of IOUtils::copy
```

Comment:
This TODO item can be removed. It is just cluttering things up, and I've been meaning to take it out. (Though feel free to leave it in, if you prefer)

Reply:
Do you think this is invalid? I thought it was still useful…

Reply:
Sure, we can keep it here as a reminder